### PR TITLE
python311Packages.anthropic: 0.19.1 -> 0.25.6

### PR DESCRIPTION
**Pull request by natsukium**
Changelog: https://github.com/anthropics/anthropic-sdk-python/releases/tag/v0.25.6

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 307401` at 6da08f5f run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/natsukium/459930e1e332bbefff0e5d3796cf2e59">python311Packages.anthropic</a></li>
    <li><a href="https://gist.github.com/natsukium/aa3a6ccff0fdb056761d16807619bbec">python312Packages.anthropic</a></li>
  </ul>
</details>

**Comment by de11n** [comment]
0.25.7 is now out.

Also it looks like we need to disable `test_copy_build_request` because it flags a memory leak. There may be actually be a memory leak, but we likely aren't going to be the ones to solve it, and they are releasing the version anyhow.

**Comment by de11n** [comment]
Oh that's already disabled but the attribute name is `disableTests` instead of `disabledTests`.

File: pkgs/development/python-modules/anthropic/default.nix [hunk 1]
```diff
@@ -56,11 +64,20 @@ buildPythonPackage rec {
...
+  pytestFlagsArray = [
+    "-W ignore::DeprecationWarning"
+  ];
+
   disabledTestPaths = [
     # require network access
     "tests/api_resources"
   ];
 
+  disableTests = [
```
Comment (de11n) [inline pkgs/development/python-modules/anthropic/default.nix:76]:
```suggestion
  disabledTests = [
```

File: pkgs/development/python-modules/anthropic/default.nix [hunk 2]
```diff
@@ -56,11 +64,20 @@ buildPythonPackage rec {
...
+    # require network access
+    "test_copy_build_request"
```
Comment (de11n) [inline pkgs/development/python-modules/anthropic/default.nix:78]: It doesn't seem like this has anything to do with network access.

```suggestion
    "test_copy_build_request" # upstream needs to fix a memory leak
```

File: pkgs/development/python-modules/anthropic/default.nix [hunk 3]
```diff
@@ -19,7 +22,7 @@
 
 buildPythonPackage rec {
   pname = "anthropic";
-  version = "0.19.1";
+  version = "0.25.6";
```
Comment (de11n) [inline pkgs/development/python-modules/anthropic/default.nix:25]: 0.25.7 is out, but that shouldn't block this merge.

**Review by de11n — CHANGES_REQUESTED** [review]

**Comment by wegank** [comment]
Already on master.

**Pull request closed without merging**